### PR TITLE
[Feature] Add private data

### PR DESCRIPTION
**Pull request by Dvergar**
This PR

## PR Guidelines

- [x] I have read the [contribution and PR guidelines](/chili-publish/studio-sdk/blob/main/CONTRIBUTING.md)

## Related tickets

https://chilipublishintranet.atlassian.net/browse/EDT-1608

### CODE REVIEW

**Review by NicolaVerbeeck — APPROVED** [review]

File: packages/sdk/src/types/VariableTypes.ts [hunk 1]
```diff
@@ -45,6 +45,7 @@ export interface Variable {
     occurrences: number;
     placeholder?: string | null;
     helpText?: string | null;
+    privateData: Record<string, string>;
```
Comment (Matthiee) [inline packages/sdk/src/types/VariableTypes.ts:48]: Maybe we should have a private data alias?

File: packages/sdk/src/controllers/VariableController.ts [hunk 2]
```diff
@@ -596,6 +596,29 @@ export class VariableController {
...
+    };
+
+    /**
+     * Gets the private data for any variable
+     * @param id the id of the variable
+     * @returns the private data
+     */
+    getPrivateData = async (id: string) => {
+        const res = await this.#editorAPI;
+        return res.getVariablePrivateData(id).then((result) => getEditorResponseData<null>(result));
```
Comment (Matthiee) [inline packages/sdk/src/controllers/VariableController.ts:619]:
```suggestion
        return res.getVariablePrivateData(id).then((result) => getEditorResponseData<PrivateData>(result));
```

**Review by Matthiee — CHANGES_REQUESTED** [review]

File: packages/sdk/src/controllers/VariableController.ts (hunk 2 above)
Comment (brapoprod) [reply packages/sdk/src/controllers/VariableController.ts:619]: should the setter also not return it?

File: packages/sdk/src/types/VariableTypes.ts (hunk 1 above)
Comment (brapoprod) [reply packages/sdk/src/types/VariableTypes.ts:48]: Could be handy

File: packages/sdk/src/controllers/VariableController.ts (hunk 2 above)
Comment (Dvergar) [reply packages/sdk/src/controllers/VariableController.ts:619]: Looking at other setters, it seems we mostly (only?) return when an id is generated from an entity we created. Here the setter is a full replace of the private data so I don't think it has much value.

File: packages/sdk/src/controllers/VariableController.ts [hunk 3]
```diff
@@ -596,6 +597,29 @@ export class VariableController {
         return this.applyPrefixSuffixDeltaUpdate(id, update);
     };
 
+    /**
+     * Sets the private data for any variable
+     * @param id the id of the variable to update
+     * @param privateData the private data
+     * @returns
+     */
+    setPrivateData = async (id: string, privateData: Record<string, string>) => {
```
Comment (Matthiee) [inline packages/sdk/src/controllers/VariableController.ts:606]:
```suggestion
    setPrivateData = async (id: string, privateData: PrivateData) => {
```

**Review by Matthiee — APPROVED** [review]
LGTM

**Review by Matthiee — APPROVED** [review]

**Pull request merged**